x/nftmarket/keeper/nft_listing.go

```diff
@@ -238,6 +231,16 @@ func (k Keeper) ListNft(ctx sdk.Context, msg *types.MsgListNft) error {
 	}
 	k.SaveNftListing(ctx, listing)
 
+	// get the memo data from Tx contains MsgListNft
+	// If memo data is empty, the hook method is not gonna be called.
+	txMemo, err := k.GetMemo(ctx)
```

Comment:
I haven't seen any memo use case in this way, wouldn't it better to add just memo field as custom field on NftListing message?

Reply:
plz answer it